File: Questao2.py
def is_fibonacci(num):
    a, b = 0, 1
    if num == 0 or num == 1:
        return True

    while b < num:
        a = b
        b = a + b

    return b == num

num = 21  # Substitua este valor pela entrada desejada

if is_fibonacci(num):
    print(f"{num} pertence à sequência de Fibonacci.")
else:
    print(f"{num} não pertence à sequência de Fibonacci.")

File: Questao2.ts
// // # Fibonacci

import prompt from "prompt-sync";

function Fibonacci(num: number): boolean {
  let a: number = 0;
  let b: number = 1;
  let temp: number = 0;

  if (num === 0 || num === 1) return true;

  while (b < num) {
    temp = a + b; // 1 // 2 // 3
    a = b; // a = 1 // a = 1 // a = 2
    b = temp; // b = 1 // b = 2 // b = 3
  }

  return b === num;
}

{
  /* 

    // Se quiser inserir a variavel pelo teclado

    let resposta = prompt();
    const num = resposta("Digite um valor: ");
    console.log(`Você digitou: ${num}`);
    
*/
}

let num: number = 21;

if (Fibonacci(num)) {
  console.log(`${num} pertence à sequência de Fibonacci.`);
} else {
  console.log(`${num} não pertence à sequência de Fibonacci.`);
}
